tests/framework/extensions/provisioning/verify.go

```diff
@@ -335,3 +342,32 @@ func GetSnapshots(client *rancher.Client, localclusterID string, clusterName str
 	return snapshots, nil
 
 }
+
+// VerifySSHTests validates the ssh tests listed in the config on each node of the cluster
+func VerifySSHTests(t *testing.T, client *rancher.Client, clusterObject *steveV1.SteveAPIObject, clusterSpec *provv1.ClusterSpec) {
```

Comment:
Why pass the clusterSpec, pass the list of sshTests instead. You really shouldn't be modifying the native k8s object for this. This could mess up rancher